### PR TITLE
CircleCI changes for build, push and deploy of NH to staging and prod

### PR DESCRIPTION
**Pull request by zolakeith**
This commit updates the CircleCI configuration so that the docker image is pushed to Zola's AWS ECR registry.

This is achieved by changing the default dockerhub references to Zola's aws registry.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -60,26 +60,31 @@ docker_import: &docker_import
...
-      docker push \
-        nerveshub/$APP_NAME:$DOCKER_TAG
+      aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/o0z1n0s8
+      docker push public.ecr.aws/o0z1n0s8/nerveshub/$APP_NAME:$DOCKER_TAG
+
+docker_login: &docker_login
+  aws-cli/setup:
+    aws-access-key-id: AWS_ACCESS_KEY
+    aws-secret-access-key: AWS_ACCESS_SECRET
+    aws-region: AWS_REGION_NAME
```
Comment (zolakeith) [inline .circleci/config.yml:73]: It feels like "docker_login" is the wrong name for this. Perhaps "awscli_setup" would be a better name?

File: .circleci/config.yml [hunk 2]
```diff
@@ -60,26 +60,31 @@ docker_import: &docker_import
...
-        nerveshub/$APP_NAME:$DOCKER_VERSION_TAG
+        public.ecr.aws/o0z1n0s8/nerveshub/$APP_NAME:$DOCKER_VERSION_TAG
       docker push \
-        nerveshub/$APP_NAME:latest
+        public.ecr.aws/o0z1n0s8/nerveshub/$APP_NAME:latest
 
 version: 2.1
 orbs:
   docker: circleci/docker@2.0.3
+  aws-cli: circleci/aws-cli@2.0
```
Comment (zolakeith) [inline .circleci/config.yml:87]: I think [circleci/aws-cli@3.1.3](https://circleci.com/developer/orbs/orb/circleci/aws-cli) is the latest orb - any reason not to use it?

File: .circleci/config.yml [hunk 3]
```diff
@@ -294,50 +302,50 @@ workflows:
         #       tags:
         #         only: /.*/
         - build-www:
-            context: org-global
+            context: aws
```
Comment (zolakeith) [inline .circleci/config.yml:305]: What is this "context" and why did we change it? Does it mean anything?

**Review by zolakeith — COMMENTED** [review]
@slow-1oris I've made these changes into a separate PR here where we can discuss and refine, then squash into one commit. I think it might make sense to keep main as "clean" as possible, to make sync'ing with upstream easier. Let me know what you think.

File: .circleci/config.yml [hunk 4]
```diff
@@ -28,9 +28,9 @@ docker_build_release: &docker_build_release
     name: Build docker images
     command: |
       docker build \
-        -t nerveshub/$APP_NAME:$DOCKER_TAG \
-        -t nerveshub/$APP_NAME:$DOCKER_VERSION_TAG \
-        -t nerveshub/$APP_NAME:latest \
+        -t public.ecr.aws/o0z1n0s8/nerveshub/$APP_NAME:$DOCKER_TAG \
+        -t public.ecr.aws/o0z1n0s8/nerveshub/$APP_NAME:$DOCKER_VERSION_TAG \
+        -t public.ecr.aws/o0z1n0s8/nerveshub/$APP_NAME:latest \
```
[the remark this answers is not in the document]
Comment (zolakeith) [reply .circleci/config.yml:33]: I do wonder though, if we are "logging in" with our AWS credentials, won't this `public.ecr.aws/o0z1n0s8` path be default, and so perhaps we can remove it?

[the remark this answers is not in the document]
Comment (zolakeith) [reply .circleci/config.yml:33]: I might be wrong here, but can we configure docker to use our aws registry somehow? https://docs.docker.com/registry/configuration/

[the remark this answers is not in the document]
Comment (zolakeith) [reply .circleci/config.yml:33]: Ignore me, looks like it's not possible to set ENV vars - it needs to be configured in the docker deamon json, so passing in the url as we're doing here is the "default" way.

File: .circleci/config.yml (hunk 3 above)
Comment (zolakeith) [reply .circleci/config.yml:305]: Is this a CircleCI thing? I see `context: aws` is also used in these examples: https://circleci.com/developer/orbs/orb/circleci/aws-cli

File: .circleci/config.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (slow-1oris) [reply .circleci/config.yml:33]: > I do wonder though, if we are "logging in" with our AWS credentials, won't this `public.ecr.aws/o0z1n0s8` path be default, and so perhaps we can remove it?

we are basically  logging into 'all public repos', so unfortunately we have to specify the $DOCKER_REGISTRY

File: .circleci/config.yml (hunk 3 above)
Comment (slow-1oris) [reply .circleci/config.yml:305]: a context is like a namespace for environment variables. 
'org-global' is apparently the context NH has in their repo setup. 
we don't have 'org-global', 
'aws' context was created by me to hold parameters related to AWS

Comment (zolakeith) [reply .circleci/config.yml:305]: Ok makes sense, thanks :+1:

**Comment by zolakeith** [comment]
Thanks for the updates @slow-1oris, I'll squash this and merge...

**Comment by slow-1oris** [comment]
@zolakeith 
I have changed this PR to point to 'develop' branch.
I've also added the deploy step and would like to squash and merge into develop to test the deployment step

**Comment by zolakeith** [comment]
@slow-1oris 100% agree - do you want to squash? If so, I think one commit with well written commit message would be helpful, so I would recommend rebase and squash locally, rather than using Github UI

**Comment by zolakeith** [comment]
Otherwise I can squash if you prefer.

**Pull request merged**